### PR TITLE
Dev item editor

### PR DESCRIPTION
**Pull request by swordmaster2k**
Implementation of the JsonItemSerializer. 

Can ignore some file changes they were picked up after a rebase from develop, they've already been reviewed.

### CODE REVIEW

File: src/main/java/net/rpgtoolkit/common/assets/Item.java [hunk 1]
```diff
@@ -242,6 +243,15 @@ public void setUsersSpecified(boolean usersSpecified) {
   }
 
   /**
+   * 
+   * @param userChar 
+   */
+  public void setUserChar(List<String> userChar) {
```
Comment (goblinJoel) [inline src/main/java/net/rpgtoolkit/common/assets/Item.java:249]: I think we should probably consistently pluralize our collection names and the functions that go with them. setUserChars, equipLocations, etc.

File: src/test/java/net/rpgtoolkit/common/assets/serialization/AssetSerializerTestHelper.java [hunk 2]
```diff
@@ -1,58 +1,63 @@
...
+import net.rpgtoolkit.common.assets.resources.ResourceAssetHandle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * 
+ * @author Joel Moore
```
Comment (goblinJoel) [inline src/test/java/net/rpgtoolkit/common/assets/serialization/AssetSerializerTestHelper.java:25]: Going by the file history (and not remembering writing this), I think this is actually by Euix.

File: src/test/java/net/rpgtoolkit/common/assets/serialization/legacy/TestSpecialMoveSerializer.java [hunk 3]
```diff
@@ -1,61 +1,68 @@
...
+import net.rpgtoolkit.common.assets.serialization.AssetSerializerTestHelper;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+
+/**
+ *
+ * @author Joel Moore
```
Comment (goblinJoel) [inline src/test/java/net/rpgtoolkit/common/assets/serialization/legacy/TestSpecialMoveSerializer.java:22]: This one also is by Euix, I think.

**Review by goblinJoel — COMMENTED** [review]
Didn't find anything major, which is good, because I didn't get around to reviewing it until now.

File: src/test/java/net/rpgtoolkit/common/assets/serialization/AssetSerializerTestHelper.java (hunk 2 above)
Comment (swordmaster2k) [reply src/test/java/net/rpgtoolkit/common/assets/serialization/AssetSerializerTestHelper.java:25]: Will update, my mistake.

**Pull request merged**